src/plugins/data_source_management/public/plugin.ts

```diff
@@ -110,6 +116,7 @@ export class DataSourceManagementPlugin
       ui: {
         DataSourceSelector: createDataSourceSelector(uiSettings, dataSource),
         getDataSourceMenu: <T>() => createDataSourceMenu<T>(),
+        dataSourceURLKey: DATA_SOURCE_URL_KEY,
```

Comment:
Hi Huy. Could you help me to understand what does this  dataSourceURLKey do and how does it help plugin team?

Reply:
Before, plugins had to manage how they wanted to persist the selected datasource across their pages. Most opted for the datasource to be a part of the url since it makes the page bookmark-able and easier than drilling down props. However, this urlkey `&dataSource=` is not global, meaning different plugins can have different urlkeys. This PR will expose a common key `dataSource` for them to consume to increase consistency among plugin urls.

Reply:
Got it. Thanks for explaining it.